homeassistant/components/mqtt/sensor.py

```diff
@@ -290,6 +290,7 @@ def _update_last_reset(msg: ReceiveMessage) -> None:
         @log_messages(self.hass, self.entity_id)
         def message_received(msg: ReceiveMessage) -> None:
             """Handle new MQTT messages."""
+            self.monitor.track({"_attr_native_value", "_attr_last_reset"})
```

Comment:
This will store the current value of these attributes, so the can be evaluated for changes when we are about to decide if the state should be written.